src/Servers/Kestrel/Kestrel/src/WebHostBuilderKestrelExtensions.cs

```diff
@@ -33,6 +33,7 @@ public static IWebHostBuilder UseKestrel(this IWebHostBuilder hostBuilder)
             {
                 // Don't override an already-configured transport
                 services.TryAddSingleton<IConnectionListenerFactory, SocketTransportFactory>();
+                services.TryAddSingleton<ISocketConnectionContextFactory, SocketConnectionContextFactory>();
```

Comment:
I don't think we need to register this as a service in UseKestrel(). This is for other custom transports (IConnectionListenerFactory implementations) to use, not us.

Reply:
Though this does ensure we have decent test coverage. I'm just wary of adding new extensibility points. Maybe it's fine.

Reply:
The more I think about this, the more I think the extra test coverage is not a reason to do this. Most of the logic is in SocketConnection which is shared anyway. We can add tests specifically for any interesting behavior in SocketConnectionContextFactory if need be.